returnn/util/file_cache.py

```diff
@@ -99,7 +106,16 @@ def get_file(self, src_filename: str) -> str:
         :return: cached file path (in the cache directory)
         """
         dst_filename = self._get_dst_filename(src_filename)
-        self._copy_file_if_needed(src_filename, dst_filename)
+        last_error = None
+        for num_try in range(1, self._num_tries + 1):
+            try:
+                self._copy_file_if_needed(src_filename, dst_filename)
+                break
+            except Exception as e:
```

Comment:
This is way too generic. You should only catch very specific errors, e.g. out-of-disk space or so.

Reply:
Are you sure? This is just for retries anyways, and if the error is not transient but persists will be re-thrown at the end.

Reply:
I'm also not sure we know the set of errors upfront or if the errors we get are even meaningful. Out of disk space, for example, is just an `OSError`.

Reply:
But we should not retry if there is some other exception. There could be all kind of strange things happening otherwise.



> Out of disk space, for example, is just an `OSError`.

But you can just check the error code for out-of-disk-space. Should be `errno.ENOSPC`.

Reply:
... and then there are a bunch of other transient errors that we want to retry on, no? I'm just arguing trying to make this list exhaustive is not worth it when we're re-raising on the last try anyway.

Reply:
> there are a bunch of other transient errors that we want to retry on, aren't there?

No, that's exactly my point. We should not just retry on whatever problem there is.